x/src/cargo.rs

```diff
@@ -98,27 +127,43 @@ impl Cargo {
     }
 
     fn do_run(&mut self, log: bool) -> Result<Output> {
+        //these are arguments on the cargo command
+        if !self.direct_args.is_empty() {
+            self.inner.args(&self.direct_args);
+        }
+
+        //this are arguments are passed through cargo to underlaying
+        //executable
         if !self.pass_through_args.is_empty() {
             self.inner.arg("--").args(&self.pass_through_args);
         }
+
         // once all the arguments are added to the command we can log it.
         if log {
+            &self
+                .env_additions
+                .iter()
+                .for_each(|t| info!("Env {:?}: {:?}", t.0, t.1));
             info!("Executing: {:?}", &self.inner);
         }
+        self.inner.envs(&self.env_additions);
         let now = Instant::now();
         let output = self.inner.output()?;
-        // once all the command has been executed we log it's success or failure.
+        // once the command has been executed we log it's success or failure.
         if log {
-            info!(
-                "{} in {}ms: {:?}",
-                if output.status.success() {
-                    "Completed"
-                } else {
-                    "Failed"
-                },
-                now.elapsed().as_millis(),
-                &self.inner
-            );
+            if output.status.success() {
+                info!(
+                    "Completed in {}ms: {:?}",
+                    now.elapsed().as_millis(),
+                    &self.inner
+                );
+            } else {
+                warn!(
```

Comment:
different coloring, maybe a different pr, only started matter to me with --no-fail-fast

Reply:
it's been helpful to visually parse logs when consensus tests fail in the full test run.  

I didn't see anything like log(Level, "{}", str), but would have preferred that had it been available.

Reply:
Exactly that exists: https://docs.rs/log/0.4.10/log/macro.log.html